pyscriptjs/src/pyodide.ts

```diff
@@ -97,6 +97,32 @@ export class PyodideRuntime extends Runtime {
         }
     }
 
+    /**
+     *
+     * @param path : the path in the filesystem
+     * @param fetch_path : the path to be fetched
+     *
+     * For a given resource with `fetch_path` as `http://dummy.com/hi.py`
+     * and the location of `path` in the FS to be `a/b/c/foo.py`
+     *
+     * Iteratively analyses the paths:
+     *  - `a`
+     *  - `a/b`
+     *  - `a/b/c`
+     *
+     * The analysis returns if the path exists and if it's parent directory exists
+     * If the path itself doesn't exist, we create it
+     *
+     * Due to the manner in which we proceed, the parent will ALWAYS exist.
+     *
+     * Thus, the iteration proceeds in the following manner:
+     *
+     * - `a` doesn't exist but it's parent i.e. `root` exists --> create `a`
+     * - `a/b` doesn't exist but it's parent i.e. `a` exists --> create `a/b`
+     * - `a/b/c` doesn't exist but it's parent i.e. `a/b` exists --> create `a/b/c`
+     *
+     * Finally, write content of `http://dummy.com/hi.py` to `a/b/c/foo.py`
+     */
```

Comment:
Can you also add comments? As a consumer of the code, I'd like a comment that saw:

* what does `const { exists, parentExists } = this.interpreter.FS.analyzePath(eachPath);` and the lines `130 to 138` are doing (basically, recursively create the path on the FS)
* `const response = await robustFetch(fetch_path);` is the response being checked? What happens if it fails to fetch?
* anything in particular that should be a comment on how `const stream = this.interpreter.FS.open(pathArr.join('/'), 'w');` works?

Reply:
done